src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -150,4 +150,11 @@ public List findByAuthor(@PathVariable String bookAuthor) {
         return ResponseEntity.status(HttpStatus.CREATED)
             .body(openLibraryService.saveBook(openLibraryBook));
     }
+
+    @GetMapping("/find_by_publisher_genre_and_year")
```

Comment:
To use more restful method names we can avoid to set this URL and used something like this:
```
@GetMapping
@RequestMapping(params = {"publisher", "genre", "year"})
public List<Book> findByPublisherAndGenreAndYear(
        @RequestParam(name = "publisher") String publisher,
        @RequestParam(name = "genre") String genre, @RequestParam(name = "year") String year) {
```

So, for example, this will be consumed like: `GET localhost/api/books?author=King&publisher=Editorial`
Using `@RequestMapping` we can use the same URI in different controller methods (I mean, you'll have this one and the one that get all the books) but with different kind of params :)